src/main/java/org/jitsi/videobridge/cc/vp8/VP8AdaptiveTrackProjectionContext.java

```diff
@@ -507,52 +536,59 @@ public PayloadType getPayloadType()
      * Rewrites the RTP packet that is specified as an argument.
      *
      * @param rtpPacket the RTP packet to rewrite.
-     * @param incomingPacketCache the packet cache to pull piggy-backed
-     * packets from. Null is permissible, but in that case no packets will be
-     * piggy backed.
-     * @return any RTP packets to piggy-bac
      * @throws RewriteException if a VP8 frame projection is not found
      * for the RTP packet that is specified as a parameter.
      */
     @Override
-    public VideoRtpPacket[] rewriteRtp(
-        @NotNull VideoRtpPacket rtpPacket, PacketCache incomingPacketCache)
+    public void rewriteRtp(
+        @NotNull VideoRtpPacket rtpPacket)
         throws RewriteException
     {
         if (!(rtpPacket instanceof Vp8Packet))
         {
             logger.info("Got a non-VP8 packet.");
-            return null;
+            throw new RewriteException();
```

Comment:
Seems like some context in these exceptions (of what exactly went wrong) would be useful.

Reply:
Added a message.